src/services/PostService.ts

```diff
@@ -21,30 +21,56 @@ export class PostService {
     this.transactions = new TransactionsManager(entityManager);
   }
 
-  public async getAllPosts(): Promise<PostModel[]> {
+  public async getAllPosts(user: UserModel): Promise<PostModel[]> {
     return this.transactions.readOnly(async (transactionalEntityManager) => {
       const postRepository = Repositories.post(transactionalEntityManager);
-      // filter out posts from inactive users
-      return (await postRepository.getAllPosts()).filter((post) => post.user?.isActive);
+      const activeUserPosts = (await postRepository.getAllPosts()).filter((post) => post.user?.isActive);
+      const userRepository = Repositories.user(transactionalEntityManager);
+      const userWithBlockedInfo = await userRepository.getUserWithBlockedInfo(user.id)
+      const blockedUsers = userWithBlockedInfo?.blocking;
+      const unblockedPosts = activeUserPosts.filter((post) => {
+        if (blockedUsers) {
+          for (const blockedUser of blockedUsers) {
+            if (post.user?.id == blockedUser.id) return false;
+          }
+        }
+        return true;
+      });
+      return unblockedPosts; 
```

Comment:
It looks like this code is duplicated in some of the other functions, it'd be a good idea to factor this out into a separate helper method called something like `filterBlockedUserPosts()`, which takes in a set of posts and a user id and then returns a filtered version of the posts. You can then just take whatever posts a given route was originally and pass them through this function -- i.e., if a service was originally returning `posts` which you fetched from the DB, now just return `filterBlockedUserPosts(posts, user.id)`. This should make your code easier to change in the future and avoid the duplication